File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {
    private final int CarsPerSpeed = 221;

    public double productionRatePerHour(int speed) {
        var successRate = .77;
        if (speed < 5) {
            successRate = 1;
        } else if (speed < 9) {
            successRate = .9;
        } else if (speed < 10) {
            successRate = .8;
        } 
        return speed * CarsPerSpeed * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        return (int) productionRatePerHour(speed) / 60;
    }
}

File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

let BaseCarsPerHour = 221.0

let successRate = function 
    | 1 | 2 | 3 | 4 -> 1.0
    | 5 | 6 | 7 | 8 -> 0.9
    | 9 -> 0.8
    | 10 -> 0.77
    | _ -> 0.0

let productionRatePerHour (speed: int): float = (successRate speed) * (speed |> float) * BaseCarsPerHour

let workingItemsPerMinute (speed: int): int = (productionRatePerHour speed) / 60.0 |> int
    
